Fix for [JIRA: CLIENTS-654]

empovit opened the pull request:
ListBuckets ignored the bucket type member when building a request, which made it work with the default bucket type only. Passing bucket type if not null to enable listing non-empty buckets of types other than 'default'.

empovit committed:
Pass bucket type if not null to enable listing non-empty buckets of types other than 'default'

lukebakken commented on src/main/java/com/basho/riak/client/api/commands/buckets/ListBuckets.java:
```diff
@@ -85,6 +85,11 @@ private ListBucketsOperation buildCoreOperation()
             builder.withTimeout(timeout);
         }
+
+        if (type != null) {
```
Could you please match the brace style in this file? Thank you!

empovit replied:
Sure. Should have done it, sorry

lukebakken replied:
It's no problem, I'm just being pedantic. Thank you!

empovit replied:
@lukebakken 
You're absolutely right, I have to stick to the style in use.

lukebakken commented:
Thank you for this pull request :smile: 

There should also be a unit test confirming this fix here (I think):

https://github.com/basho/riak-java-client/tree/develop/src/test/java/com/basho/riak/client/api/commands

@alexmoore - is that the right place?

empovit commented:
@lukebakken 
Ok, I see a unit test for ListBuckets now. Will try to fix it because apparently it didn't catch the bug.

lukebakken commented:
Great!

alexmoore commented:
Yep, that would be the folder to put a unit test in.

empovit committed:
ListBuckets - Fixed code style and added unit tests

empovit commented:
Fixed, added unit tests. Please review.

BTW, I think that copying the bucket type from the command context into response is wrong, and it should be returned from the backend. See com.basho.riak.client.api.commands.buckets.ListBuckets#executeAsync. As a side note, it posed a challenge when writing the unit tests, because validating an expected bucket type against a response does not make sense as the type is always the one from request.

alexmoore commented:
Yeah, our messages only return the bucket names: https://github.com/basho/riak_pb/blob/develop/src/riak_kv.proto#L127-L130

I think the original authors did this to save message space, otherwise we'd send a lot of duplicate info back with every bucket.   This applies to the RpbListKeysResp message as well. :/

alexmoore commented:
Looks good, thanks for the PR!
